README.md
# Ads Value
Replace online ads with slogans personalized for you
![ads value](https://www.bettaverse.xyz/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fadsvalue1.eac99e4c.jpg&w=1920&q=75)

##### Why?

Most people believe advertising doesn't work or, at least, it doesn't work on them.

That is the genius of ads.

If we collectively acknowledged the effectivness of advertising we would ban it immediately.

##### Reclaim your attention

Maybe you're sick of being influenced by faceless executives and would prefer to influence yourself?

Now you can replace online ads with your own slogans to positively influence your behaviour.


# Installation

1. [Download](https://github.com/bettaverse/adsvalue/archive/refs/heads/main.zip) the extension from here.
2. Click on the hambuger menu 三 in the top right Google Chrome
3. Click Extensions
4. Turn on Developer mode (in the top right)
5. Click Load unpacked
6. Select the Ads Value folder
7. Done

#### Watch this short vid:
[![Ads value youtube video](https://img.youtube.com/vi/Hu8LUbTmBtM/0.jpg)](https://www.youtube.com/watch?v=Hu8LUbTmBtM)


More info at [Bettaverse](https://bettaverse.xyz/adsvalue).





# This Extension is based on CatBlock

Thanks to the CatBlock team for their hard work. 'Ads Value' is based heavily on their code.
Here's some info and links about that project:

[Catblock](https://github.com/CatBlock/catblock) on Github

# CatBlock

## Overview

CatBlock(previously known as "AdBlock with CatBlock") is like an ad blocking extension, but instead of blocking ads it shows you pictures of cats by default (configurable).

You can also disable replacing ads by cats, so Ads Valuecan work as a standard ad blocking extension (useful for when you're doing presentations anywhere!)

If you want to find out the history of this extension, you can check it out [here](https://github.com/CatBlock/catblock/wiki/History).



## Supported browsers

Ads Valueis compatible with following browsers:

- Chrome: 49+
- Opera: 35+
- Safari: 6+
- Firefox: 48+
- Edge: Windows 10 Anniversary - version 1607





CHANGELOG.txt
## Versions ##

1.5.1:
- Block Facebook suggested posts (#120)
- Updated jQuery and jQuery UI (#130)
- Fixed not working "Report an ad" page (#126)
- Fixed not working "Block an ad" wizard (#124)
- Removed eval statements (#119)

1.5:
- Change chrome.extension.getBackgroundPage to chrome.runtime.getBackgroundPage (#118)
- Remove "Project C.A.T.S." code (#117)
- Get build info from chrome.management API (#113)
- parseURI v2 - use standardized URL constructor (#116)
- Update the i18n for Options page title (#114)
- Options page redesign (#111)
- Update libraries (#110)
- Use ES2015 classes instead of having prototype-based functions (#73)
- Refactor build script (#76)
- Pause state should use localStorage instead of sessionStorage (#100)
- Hide commands when running Setup.cmd (#109)
- All parameters are mandatory in `initEvent` (#102)

1.4.9:
- Fixed initialization of "Project CATS" for existing users (#88, #89)

1.4.8:
- Include js/cats.js script in BG scripts on Safari (#84, #87)
- Fixed setting width and height of a Safari popup (#85, #86)

1.4.7:
- Added "Project CATS" mode for C.A.T.S. backers (#82)
- Enable "Project CATS" only on Chrome, Opera and Firefox (#83)
- Update jQuery to 3.1 (#72)
- Fixed bug, which has resulted in displaying more cats than intended (#37, #79)
- Fixed a regression introduced in 1.4.6, which has resulted in not replacing ads by cats (#77, #78)
- Fixed disabling Flickr channels; they were automatically enabled after a browser restart (#74, #75)
- Fixed building Ads Valuefor Firefox (#80)

1.4.6:
- Name change of popover on Safari from "AdBlock" to "CatBlock" (#59)
- Drop support for Safari 5.x (#61)
- Change references to Ads Valueon Safari (#62)
- Unified code style (#63)
- Fixed throwing an error when "getAttribute" property of an element is undefined (#66)
- Updated translations

1.4.5:
- Add checks for undefined tabs, while getting info about them (#56)
- Simplified code for getting tab info in popup (#56)
- Show step for disabling extensions on Edge and Firefox (again) (#56)
- Removed function for normalizing requests (#56)
- Check for "message" and "stack" property when an error occurs (#58)

1.4.4:

- Adjusted steps for Firefox and Edge on ad-report page (#53)
- Changed name from "AdBlock with CatBlock" to "CatBlock" (#54)
- Adjusted minimum versions of browsers needed for running Ads Value(#54)
- Don't show how-to hide the extension button on Firefox (#53)
- Update description of the extension (#54, #55)
- Updated translations (#53)
- Fixed detection of Ads Valueon ad-report page (#53)
- Fixed using chrome.permissions on ad-report page (#53)

1.4.3:
- Fixes injecting YouTube channel logic file (#52)

1.4.2:
- Switched to asynchronous XHR throughout the code (#46)
- Updated jQuery UI library (#46)
- Fixed injecting of jQuery UI (#46)
- Fixed undoing added hiding filters within popup menu on Firefox (#46)
- Punycode library wasn't defined in Safari (#46)
- Fixed path to bandaids.js (#50)
- Moved inline CSS into .css file for an Adreport page (#48)

1.4.1:
- Added context menu options (Pause/Unpause ads value; Options page)
- Added support for Firefox browser
- Move from extension.getURL to runtime.getURL API
- Move from the deprecated extension.sendRequest/onRequest to runtime.sendMessage/onMessage
- Updates to the port.js file (reflecting the API changes)
- Updated jQuery and jQuery UI libraries
- Updated punycode.js library
- Changed Options page behaviour
- Updated translations
- Fixed selecting certain options on General tab

1.4:
- Added support for Microsoft Edge browser
- Added update checker for Edge
- Updated the AdBlock core to version 2.56
- Updated translations

1.3:
- Update the AdBlock core to version 2.50
- Add the option to "paws" the image replacement functionality
- Improve ad replacement functionality on Chrome

1.2.1:
- Minor fixes for Chrome and Opera stores

1.2:
- Updated translations
- Updated design from AdBlock 2.18
- Whitelisted icon has been improved

1.1.2:
- Bumped required minimal version of Chrome from 23 to 41
- Fixed icons on Safari
- Fixed loading of jQuery on Safari
- Removed unused icons and /filters folder

1.1.1:
- Minor fixes for getting extension published on Chrome Web Store and
  Opera Extensions Store

1.1:
- First working release of CatBlock
- One codebase for all versions of the extension
- CatBlock's settings page has been added to Options
- Updated AdBlock files to 2.7

1.0:
- Imported code from ads value.getadblock.com


tools/make_release.py
#!/usr/bin/env python3

"""
This script generates a release versions of Ads Valuefor different platforms
and also an unpacked development environment

Compatible with Python v3.x only

Author: Tomáš Taro
License: GNU GPLv3
"""

def main():
    import json # Provides JSON-related functions
    import os # Provides file-system functions
    import sys # Provides system functions
    import shutil # Provides folders functions
    import argparse # Provides functions for parsing arguments

    # Parse passed arguments
    parser = argparse.ArgumentParser(description="This script generates Ads Valueextension for different platforms and/or an unpacked development environment.")
    parser.add_argument("-b", "--browser", help="What browser do you want to build an extension for?", required=True)
    parser.add_argument("-ext", "--extension", help="Generate browser-specific extension package?", required=False)
    parser.add_argument("-devenv", help="Generate an unpacked development environment?", action="store_true")
    parser.add_argument("--updatelangs", help="If enabled, language files will be updated", action="store_true")
    args = parser.parse_args()

    # Function intended for printing bold strings
    def printBold(string):
        print("\033[1m" + string + "\033[0m")

    # An invalid option was selected, exit the script
    if args.browser != "firefox" and args.browser != "chrome" and args.browser != "opera" and args.browser != "edge" and args.browser != "all":
        printBold("You have selected an unsupported browser. Please, try again.")
        sys.exit(1)

    # When neither -ext nor -devenv has been defined, print an exception and exit the script
    if args.extension == None and args.devenv == False:
        printBold("You must specify, whether to create an unpacked development environment or to generate an extension package.")
        sys.exit(1)

    # Remove .DS_Store files
    os.system("find . -name '*.DS_Store' -type f -delete")

    # Update AB & CB language files
    if args.updatelangs == True:
        os.system("tools/update_i18n.py")

    # Remove already unused language files,
    # which have been merged into main language files
    if os.path.exists("catblock/_locales"):
        shutil.rmtree("catblock/_locales")

    # If /builds directory doesn't exist, create it
    if not os.path.exists("builds"):
        os.makedirs("builds")

    # Prepare manifest.json file depending on the browser
    def prepareManifestFile(browser):
        browserData = {
            "firefox": {
                "path": "catblock_firefox/manifest.json",
                "key": {
                    "applications": {
                        "gecko": {
                            "id": "catblock@ads value.tk",
                            "strict_min_version": "50.0"
                        }
                    },
                    "options_ui": {
                        "page": "options/index.html",
                        "open_in_tab": True
                    }
                }
            },
            "edge": {
                "path": "catblock_edge/catblock/manifest.json",
                "key": {
                    "minimum_edge_version": "38.14342.1000.0"
                }
            },
            "chrome": {
                "path": "catblock_chrome/manifest.json",
                "key": {
                    "minimum_chrome_version": "58"
                }
            },
            "opera": {
                "path": "catblock_opera/manifest.json",
                "key": {
                    "minimum_opera_version": "45"
                }
            }
        }

        # Apply platform-specific changes
        with open(browserData[browser]["path"], "r+") as browser_manifest:
            keys = json.load(browser_manifest) # Creates a dict of all messages in the AdBlock file

            # Update manifest file with browser-specific key
            keys.update(browserData[browser]["key"])

            # Needed for determining an extension ID used for automated tests via BrowserStack
            if browser == "chrome" and os.environ.get("TRAVIS") != None:
                keys.update({ "key": os.environ.get("EXTENSION_KEY") })
                # Update content security policy in order to correctly run unit tests
                keys.update({ "content_security_policy": keys["content_security_policy"] + " script-src 'unsafe-eval';" })
            elif browser == "firefox":
                keys.pop("optional_permissions", None)
                keys.pop("options_page", None)
                keys.pop("incognito", None)

            # Delete the current content of the manifest
            browser_manifest.seek(0)
            browser_manifest.truncate()

            # Re-create the manifest but with added browser-sepcific key
            browser_manifest.write(json.dumps(keys, sort_keys=True, indent=2, separators=(",", ": ")))

            browser_manifest.close()

    # End of prepareManifestFile

    # Selenium testing - enabled only on Travis-CI
    def runUnitTests(browser):

        # Currently, we are not running tests for other browsers than Chrome
        if browser != "Chrome":
            return

        # Making catblock-chrome.zip file available in PRs coming from forks
        shutil.make_archive("catblock-chrome", "zip", "catblock_chrome")

        # BS_API key is not exported in build process for forks
        if os.environ.get("BS_API") == None and os.environ.get("TRAVIS") != None:
            return print("  - Running QUnit tests on BrowserStack... N/A in PRs coming from forks.")

        if os.environ.get("TRAVIS") != None:

            from selenium import webdriver
            from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
            import time

            print("  - Running QUnit tests on BrowserStack...", end="")

            chop = webdriver.ChromeOptions()
            chop.add_extension("catblock-chrome.zip")
            chop = chop.to_capabilities()

            chop["os"] = "OS X"
            chop["os_version"] = "Sierra"
            chop["browser"] = "Chrome"
            chop["browserstack.debug"] = "true"

            driver = webdriver.Remote(
                command_executor="http://" + os.environ.get("BS_USERNAME") + ":" + os.environ.get("BS_API") + "@hub.browserstack.com:80/wd/hub",
            desired_capabilities=chop)

            driver.get("chrome-extension://mdcgnhlfpnbeieiiccmebgkfdebafodo/tools/tests/test.html")

            time.sleep(2)

            failure = driver.execute_script("return failure")

            if failure == True:
                print(driver.execute_script("return messages"))

            driver.quit()

            if failure == True:
                print("Failed!")
                sys.exit(1)

            print("Done!")
    # End of runUnitTests

    def generateBrowserPackage(browser):

        browsers = {
            "chrome": {
                "name": "Chrome",
                "path": "catblock_chrome",
                "extname": "catblock-chrome",
                "platformpkgsuffix": ".crx"
            },
            "opera": {
                "name": "Opera",
                "path": "catblock_opera",
                "extname": "catblock-opera",
                "extpkgname": "catblock-opera",
                "platformpkgsuffix": ".nex"
            },
            "firefox": {
                "name": "Firefox",
                "path": "catblock_firefox",
                "extname": "catblock-firefox",
                "platformpkgsuffix": ".xpi"
            },
            "edge": {
                "name": "Edge",
                "path": "catblock_edge",
                "extname": "catblock-edge",
                "platformpkgsuffix": "None"
            }
        }

        BROWSER_NAME = browsers[browser]["name"]
        BROWSER_PATH = browsers[browser]["path"]
        EXTENSION_NAME = browsers[browser]["extname"]
        EXTENSION_ZIP_NAME = EXTENSION_NAME + ".zip"
        EXTENSION_PKG_NAME = EXTENSION_NAME + browsers[browser]["platformpkgsuffix"]

        if BROWSER_NAME == "Chrome" and os.environ.get("TRAVIS") != None:
            print(" > Generating Ads Valuefor " + BROWSER_NAME + "...")
        else:
            print(" > Generating Ads Valuefor " + BROWSER_NAME + "...", end="")

        # If /BROWSER_PATH folder exists, delete it
        if os.path.exists(BROWSER_PATH):
            shutil.rmtree(BROWSER_PATH)

        # Copy the content of the original folder into /BROWSER_PATH
        # except of hidden files, builds and tools folders
        if BROWSER_NAME != "Edge":
            if os.environ.get("TRAVIS") != None and BROWSER_NAME == "Chrome":
                shutil.copytree(os.getcwd(), BROWSER_PATH, ignore=shutil.ignore_patterns(".*", "builds"))
            else:
                shutil.copytree(os.getcwd(), BROWSER_PATH, ignore=shutil.ignore_patterns(".*", "builds", "tools"))
        else:
            # Copy the content of the original folder into /catblock_edge/Ads Valueand ignore hidden files and builds folder
            shutil.copytree(os.getcwd(), BROWSER_PATH + "/catblock", ignore=shutil.ignore_patterns(".*", "builds"))

            # Copy instructions file and setup file into "/catblock_edge" folder
            shutil.move(BROWSER_PATH + "/catblock/tools/instructions.txt", BROWSER_PATH)
            shutil.move(BROWSER_PATH + "/catblock/tools/Setup.cmd", BROWSER_PATH)

        # Prepare manifest.json file
        prepareManifestFile(browser)

        # Run our unit tests on BrowserStack
        runUnitTests(BROWSER_NAME)

        # Generate a packed extension (either in .zip or in a browser-specific format)
        if args.extension != None:
            if BROWSER_NAME != "Chrome":
                shutil.make_archive(EXTENSION_NAME, "zip", BROWSER_PATH)
            elif BROWSER_NAME == "Chrome" and os.environ.get("TRAVIS") == None:
                shutil.make_archive(EXTENSION_NAME, "zip", BROWSER_PATH)

            # Rename to the Firefox compatible .xpi file
            # Edge doesn't support a special suffix of an extension,
            # hence we ignore -ext argument
            if args.extension == "browser" and BROWSER_NAME != "Edge":
                os.rename(EXTENSION_ZIP_NAME, EXTENSION_PKG_NAME)
                shutil.move(EXTENSION_PKG_NAME, "builds/" + EXTENSION_PKG_NAME)
            else:
                shutil.move(EXTENSION_ZIP_NAME, "builds/" + EXTENSION_ZIP_NAME)


        # When an unpacked development environment should not be created, remove the folder
        if args.devenv == False:
            shutil.rmtree(BROWSER_PATH)

        if os.environ.get("TRAVIS") == None:
            print("Done!")
        elif os.environ.get("TRAVIS") != None and BROWSER_NAME != "Chrome":
            print("Done!")

    # End of generateBrowserPackage

    # Start the magic!
    if args.devenv == True and args.extension == None:
        if args.browser != "all":
            printBold("Generating development environment for " + args.browser + "...")
        else:
            printBold("Generating development environment for all browsers...")
    elif args.devenv == True and args.extension != None:
        if args.browser != "all":
            printBold("Generating development environment and browser package for " + args.browser + "...")
        else:
            printBold("Generating development environments and browser packages for all browsers...")
    elif args.devenv == False and args.extension != None:
        if args.browser != "all":
            printBold("Generating browser package for " + args.browser + "...")
        else:
            printBold("Generating browser packages for all browsers...")

    if args.browser != "all":
        generateBrowserPackage(args.browser)
    else:
        generateBrowserPackage("chrome")
        generateBrowserPackage("opera")
        generateBrowserPackage("firefox")
        generateBrowserPackage("edge")

# Execute this script only, if it hasn't been imported as a module
if __name__ == "__main__":
    main()